docs/adding-new-resource-coverage.md

```diff
@@ -0,0 +1,269 @@
+
+```ruby
+require_relative '../resource'
+
+module Convection
+  module Model
+    class Template
+      class Resource
+        ##
+        # AWS::EC2::DHCPOptions
+        ##
+        class DHCPOptions < Resource
+          type 'AWS::EC2::DHCPOptions'
+          property :domain_name, 'DomainName'
+        end
+      end
+    end
+  end
+end
+```
+
+4. You have probably noticed that some of these parameters expect a string array. For those we will add `:type => :list`. See below
+
+```ruby
+require_relative '../resource'
+
+module Convection
+  module Model
+    class Template
+      class Resource
+        ##
+        # AWS::EC2::DHCPOptions
+        ##
+        class DHCPOptions < Resource
+          type 'AWS::EC2::DHCPOptions'
+          property :domain_name, 'DomainName'
+          property :domain_name_servers, 'DomainNameServers', :type => :list
+          property :netbios_name_servers, 'NetbiosNameServers', :type => :list
+          property :netbios_node_type, 'NetbiosNodeType'
+          property :ntp_servers, 'NtpServers', :type => :list
+        end
+      end
+    end
+  end
+end
+```
+
+5. To add tag support to the resource add the below block and include
+
+```ruby
+include Model::Mixin::Taggable
+
+def render(*args)
+  super.tap do |resource|
+    render_tags(resource)
+  end
+end
+```
+
+6. The completed class should look like the below
+
+```ruby
+require_relative '../resource'
+
+module Convection
+  module Model
+    class Template
+      class Resource
+        ##
+        # AWS::EC2::DHCPOptions
+        ##
+        class DHCPOptions < Resource
+          include Model::Mixin::Taggable
+
+          type 'AWS::EC2::DHCPOptions'
+          property :domain_name, 'DomainName'
+          property :domain_name_servers, 'DomainNameServers', :type => :list
+          property :netbios_name_servers, 'NetbiosNameServers', :type => :list
+          property :netbios_node_type, 'NetbiosNodeType'
+          property :ntp_servers, 'NtpServers', :type => :list
+
+          def render(*args)
+            super.tap do |resource|
+              render_tags(resource)
+            end
+          end
+        end
+      end
+    end
+  end
+end
+```
+
+7. Unit Tests! Adding tests for resources is simple, create a new test class. `convection/spec/convection/model/template/resource/ec2_dhcp_options_spec.rb`
+8. Set up the basic structure of the class.
+```ruby
+require 'spec_helper'
+
+class Convection::Model::Template::Resource
+  describe DHCPOptions do
+    let(:template) do
+      Convection.template do
+
+
+      end
+    end
+
+  end
+end
+```
+9. Add the mock template. Below is a example
+```ruby
+require 'spec_helper'
+
+class Convection::Model::Template::Resource
+  describe DHCPOptions do
+    let(:template) do
+      Convection.template do
+
+        ec2_dhcp_options 'TestOptions' do
```

Comment:
This actually wouldn't work right now. Currently the `dsl_name` would default to `dhcp_options` since we didn't include `EC2` in the class name. We can either do that or pass in the second parameter `dsl_name` to the `type` call in the `DHCPOptions` class.